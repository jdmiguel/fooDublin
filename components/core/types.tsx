export enum LogoSize {
  BIG = 'big',
  SMALL = 'small',
}

export enum LoaderType {
  CIRCLE = 'circle',
  LINE = 'line',
}

export enum LoaderMode {
  DARK = 'dark',
  LIGHT = 'light',
}

export type ListItem = {
  iconSrc?: string;
  id: number;
  name: string;
  path: string;
};

export enum CardType {
  SUGGESTION = 'suggestion',
  STANDART = 'standart',
  HIGHLIGHT = 'highlight',
}

export enum CustomLinkSize {
  BIG = 'big',
  SMALL = 'small',
}

export enum BreadcrumbsType {
  HOME = 'home',
  SEARCH = 'search',
  DETAIL = 'detail',
  FAVORITES = 'favorite',
}

export type BreadcrumbsData = {
  text: string;
  route: string;
  asRoute: string;
  type:
    | BreadcrumbsType.HOME
    | BreadcrumbsType.SEARCH
    | BreadcrumbsType.DETAIL
    | BreadcrumbsType.FAVORITES;
};
