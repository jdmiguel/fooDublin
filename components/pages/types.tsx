import { ReactNode } from 'react';

export type Area = { id: number; name: string; path: string; latitude: number; longitude: number };
export type Cuisine = { id: number; iconSrc: string; name: string; path: string };

export type Timing = {
  id: string;
  day: string;
  schedule: string;
};

export type RestaurantsRequestParam = number | null | string | undefined;

export type RestaurantsRequestParams = {
  latitude: number;
  longitude: number;
  cuisine?: string;
  offset?: number;
};

export type Restaurant = {
  id: string;
  imgSrc: string;
  title: string;
  content: ReactNode | string;
  route: string;
  asRoute: string;
};

export type HighlightRestaurant = Restaurant & { featuredSrc: string };

export type Coordinates = {
  latitude: number;
  longitude: number;
};

export type HourDetail = {
  is_overnight: boolean;
  start: string;
  end: string;
  day: number;
};

export type Hour = {
  open: HourDetail[];
  hours_type: string;
  is_open_now: boolean;
};

export type Location = {
  address1: string;
  address2: null;
  address3: string;
  city: string;
  zip_code: string;
  country: string;
  state: string;
  display_address: string[];
};

export type Category = {
  alias: string;
  title: string;
};

export type UserReview = {
  id: string;
  profile_url: string;
  image_url: string;
  name: string;
};

export type Review = {
  id: string;
  url: string;
  text: string;
  rating: number;
  time_created: string;
  user: UserReview;
};

export type FetchedRestaurantDetails = {
  id: string;
  alias: string;
  name: string;
  image_url: string;
  is_claimed: boolean;
  is_closed: boolean;
  url: string;
  phone: string;
  display_phone: string;
  review_count: number;
  categories: Category[];
  rating: number;
  location: Location & { cross_streets: string };
  coordinates: Coordinates;
  photos: string[];
  price: string;
  hours: Hour[];
  transactions: any[];
  reviews: Review[];
};

export type RestaurantDetails = {
  imgSrc: string;
  name: string;
  address: string;
  street: string;
  categories: string[];
  hours: HourDetail[] | null;
  rating: number;
  price: string | null;
  reviewCount: number;
  phone: string;
  reviews: Review[];
};

export type Suggestion = Pick<FetchedRestaurantDetails, 'id' | 'name'> & {
  route: string;
  asRoute: string;
};

export type FetchedRestaurant = Omit<FetchedRestaurantDetails, 'is_claimed'> & {
  distance: number;
};
