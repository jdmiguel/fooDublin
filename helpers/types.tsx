import { ReactNode } from 'react';

export type InitialAppState = {
  favorites: Restaurant[];
  relatedRestaurants: Restaurant[];
  breadcrumbs: BreadcrumbsData[];
};

// Core / ui components

export enum LogoSize {
  BIG = 'big',
  SMALL = 'small',
}

export enum LoaderType {
  CIRCLE = 'circle',
  LINE = 'line',
}

export enum LoaderMode {
  DARK = 'dark',
  LIGHT = 'light',
}

export type ListItem = {
  iconSrc?: string;
  id: number;
  name: string;
  path: string;
};

export enum CardType {
  SUGGESTION = 'suggestion',
  STANDART = 'standart',
  HIGHLIGHT = 'highlight',
}

export enum CustomLinkSize {
  BIG = 'big',
  SMALL = 'small',
}

// Restaurant data

export type Timming = {
  id: string;
  day: string;
  schedule: string;
};

export enum EntityType {
  CITY = 'city',
  SUBZONE = 'subzone',
}

export type RestaurantsRequestParams = {
  entity_id: number | null;
  cuisines: number | null;
  entity_type: EntityType.CITY | EntityType.SUBZONE;
  start?: number;
  sort?: string;
  order?: string;
  q?: string;
  count?: number;
};

export type Restaurant = {
  id: number;
  imgSrc: string;
  title: string;
  content: ReactNode | string;
  route: string;
  asRoute: string;
};

export type RawRestaurant = {
  restaurant: {
    R: {
      has_menu_status: {
        delivery: number;
        takeaway: number;
      };
      res_id: number;
      is_grocery_store: boolean;
    };
    apikey: string;
    id: number;
    name: string;
    url: string;
    location: {
      address: string;
      locality: string;
      city: string;
      city_id: number;
      latitude: number;
      longitude: number;
      zipcode: string;
      country_id: number;
      locality_verbose: string;
    };
    switch_to_order_menu: number;
    cuisines: string;
    timings: string;
    average_cost_for_two: number;
    price_range: number;
    currency: string;
    highlights: string[];
    offers: any[];
    opentable_support: number;
    is_zomato_book_res: number;
    mezzo_provider: string;
    is_book_form_web_view: number;
    book_form_web_view_url: string;
    book_again_url: string;
    thumb: string;
    user_rating: {
      aggregate_rating: number;
      rating_text: string;
      rating_color: string;
      rating_obj: {
        title: {
          text: number;
        };
        bg_color: {
          type: string;
          tint: number;
        };
      };
      votes: number;
    };
    all_reviews_count: number;
    photos_url: string;
    photo_count: number;
    menu_url: string;
    featured_image: string;
    has_online_delivery: number;
    is_delivering_now: number;
    store_type: string;
    include_bogo_offers: boolean;
    deeplink: string;
    is_table_reservation_supported: number;
    has_table_booking: number;
    events_url: string;
    phone_numbers: string;
    all_reviews: {
      reviews: [
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
      ];
    };
    establishment: string[];
  };
};

export type RestaurantDetail = {
  imgSrc: string;
  thumbSrc: string;
  name: string;
  location: string;
  cuisines: string;
  timings: string;
  rating: number;
  votes: number;
  average: string;
  establishment: string;
  highlights: string[];
  phone: string;
  address: string;
};

export type Review = {
  id: number;
  userName: string;
  userImgSrc: string;
  rating: number;
  date: string;
  text: string;
};

export type RawReview = {
  review: {
    rating: number;
    review_text: string;
    id: number;
    rating_color: string;
    review_time_friendly: string;
    rating_text: string;
    timestamp: number;
    likes: number;
    user: {
      name: string;
      foodie_level: string;
      foodie_level_num: number;
      foodie_color: string;
      profile_url: string;
      profile_image: string;
      profile_deeplink: string;
    };
    comments_count: number;
  };
};

// Breadcrumbs

export enum BreadcrumbsType {
  HOME = 'home',
  SEARCH = 'search',
  DETAIL = 'detail',
  FAVORITES = 'favorite',
}

export type BreadcrumbsData = {
  text: string;
  route: string;
  asRoute: string;
  type:
    | BreadcrumbsType.HOME
    | BreadcrumbsType.SEARCH
    | BreadcrumbsType.DETAIL
    | BreadcrumbsType.FAVORITES;
};

// Actions

type AddFavoriteAction = {
  type: 'ADD_FAVORITE';
  favorite: Restaurant;
};

type DeleteFavoriteAction = {
  type: 'DELETE_FAVORITE';
  id: number;
};

type SetRelatedRestaurantsAction = {
  type: 'SET_RELATED_RESTAURANTS';
  relatedRestaurants: Restaurant[];
};

type ClearRelatedRestaurantsAction = {
  type: 'CLEAR_RELATED_RESTAURANTS';
};

type AddBreadcrumbsAction = {
  type: 'ADD_BREADCRUMBS';
  breadcrumbs: BreadcrumbsData;
};

type ReplaceBreadcrumbsAction = {
  type: 'REPLACE_BREADCRUMBS';
  index: number;
  breadcrumbs: BreadcrumbsData;
};

export type Actions =
  | AddFavoriteAction
  | DeleteFavoriteAction
  | SetRelatedRestaurantsAction
  | ClearRelatedRestaurantsAction
  | AddBreadcrumbsAction
  | ReplaceBreadcrumbsAction;
