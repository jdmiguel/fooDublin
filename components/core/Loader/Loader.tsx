import React from 'react';
import styled, { keyframes, css } from 'styled-components';

export enum Mode {
  LIGHT = 'light',
  DARK = 'dark',
}

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const animation = css`
  animation: ${rotate} 0.8s linear infinite;
`;

type LoaderProps = {
  className?: string;
  text?: string;
  mode: Mode.LIGHT | Mode.DARK;
};

const StyledLoader = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const StyledLoaderImage = styled.div<{ mode: Mode.LIGHT | Mode.DARK }>`
  border: 2px solid
    ${({ theme, mode }) =>
      mode === Mode.LIGHT
        ? theme.palette.PRIMARY_LIGHT
        : theme.palette.PRIMARY};
  border-top: 2px solid
    ${({ theme, mode }) =>
      mode === Mode.LIGHT
        ? theme.palette.PRIMARY_MEDIUM
        : theme.palette.LIGHT_SOFT};
  border-radius: 50%;
  width: 30px;
  height: 30px;
  ${animation};
`;
const StyledLoaderText = styled.p<{ mode: Mode.LIGHT | Mode.DARK }>`
  margin-top: 8px;
  font-size: 0.85rem;
  color: ${({ theme, mode }) =>
    mode === Mode.LIGHT ? theme.palette.LIGHT_MAX : theme.palette.DARK_MAX};
`;

const Loader: React.FC<LoaderProps> = ({ className, text, mode }) => (
  <StyledLoader className={className}>
    <StyledLoaderImage mode={mode} />
    {text && <StyledLoaderText mode={mode}>{text}</StyledLoaderText>}
  </StyledLoader>
);

export default Loader;
