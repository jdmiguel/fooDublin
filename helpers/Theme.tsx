import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
