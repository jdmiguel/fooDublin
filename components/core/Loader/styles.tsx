import styled from 'styled-components';

import { circleLoaderAnimation, lineLoaderAnimation } from '../../../helpers/animations';

import { LoaderMode } from '../../core/types';

// Circle loader
export const StyledCircleLoader = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const StyledCircleLoaderImage = styled.div<{
  mode: LoaderMode.DARK | LoaderMode.LIGHT;
}>`
  border: 2px solid
    ${({ theme, mode }) =>
      mode === LoaderMode.DARK ? theme.palette.PRIMARY : theme.palette.PRIMARY_LIGHT};
  border-top: 2px solid
    ${({ theme, mode }) =>
      mode === LoaderMode.DARK ? theme.palette.LIGHT_SOFT : theme.palette.PRIMARY_MEDIUM};
  border-radius: 50%;
  width: 30px;
  height: 30px;
  ${circleLoaderAnimation};
`;

export const StyledCircleLoaderText = styled.p<{
  mode: LoaderMode.DARK | LoaderMode.LIGHT;
}>`
  margin-top: 8px;
  font-size: 0.85rem;
  color: ${({ theme, mode }) =>
    mode === LoaderMode.DARK ? theme.palette.DARK_MAX : theme.palette.LIGHT_MAX};
`;

// Line loader
export const StyledLineLoader = styled.div`
  width: 100%;
`;

export const StyledLineLoaderShape = styled.span`
  display: block;
  width: 100%;
  height: 5px;
  background: ${({ theme }) => theme.palette.PRIMARY};
  ${lineLoaderAnimation};
`;
