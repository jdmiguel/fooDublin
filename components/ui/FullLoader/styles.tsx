import styled, { css } from 'styled-components';

import { fadeInAnimation } from '@/helpers/animations';

import { LoaderType } from '../../core/types';

const FullCircleLoaderWrapperCSS = css`
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${({ theme }) => theme.palette.LIGHT_MEDIUM};
  ${fadeInAnimation};
  @media only screen and (min-width: 992px) {
    min-height: 200px;
  }
`;

export const StyledFullLoaderWrapper = styled.div<{
  isShowed: boolean;
  type: LoaderType.CIRCLE | LoaderType.LINE;
}>`
  width: 100%;
  height: calc(100% - 80px);
  position: fixed;
  z-index: 1;
  left: 0;
  top: ${({ type }) => (type === LoaderType.CIRCLE ? '80px' : 0)};
  visibility: ${({ isShowed }) => (isShowed ? 'visible' : 'hidden')};
  ${({ type }) => type === LoaderType.CIRCLE && FullCircleLoaderWrapperCSS}
`;
