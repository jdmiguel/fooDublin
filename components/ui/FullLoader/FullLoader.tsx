import { ReactNode } from 'react';
import { StyledFullLoaderWrapper } from './styles';
import { LoaderType } from '../../core/types';

type FullLoaderProps = {
  isShowed?: boolean;
  children: ReactNode;
  type?: LoaderType.CIRCLE | LoaderType.LINE;
};

export const FullLoader: React.FC<FullLoaderProps> = ({
  isShowed = true,
  children,
  type = LoaderType.CIRCLE,
}) => (
  <StyledFullLoaderWrapper isShowed={isShowed} type={type}>
    {children}
  </StyledFullLoaderWrapper>
);
