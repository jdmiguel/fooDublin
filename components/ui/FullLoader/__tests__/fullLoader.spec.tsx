/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import { FullLoader } from '../FullLoader';
import { Loader } from '../../../core/Loader/Loader';
import { renderWithTheme } from '../../../../helpers/Theme';
import { LoaderType } from '../../../core/types';

describe('Component: Loader', () => {
  it('should render as circle type and be shown', () => {
    const { container } = render(
      renderWithTheme(
        <FullLoader isShown={true}>
          <Loader />
        </FullLoader>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as circle type and be hidden', () => {
    const { container } = render(
      renderWithTheme(
        <FullLoader isShown={false}>
          <Loader />
        </FullLoader>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as line type and be shown', () => {
    const { container } = render(
      renderWithTheme(
        <FullLoader isShown={true} type={LoaderType.LINE}>
          <Loader />
        </FullLoader>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as line type and be hidden', () => {
    const { container } = render(
      renderWithTheme(
        <FullLoader isShown={false} type={LoaderType.LINE}>
          <Loader />
        </FullLoader>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
