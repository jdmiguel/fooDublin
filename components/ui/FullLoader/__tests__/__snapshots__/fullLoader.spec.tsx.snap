// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader should render as circle type and be hidden 1`] = `
.c0 {
  width: 100%;
  height: calc(100% - 80px);
  position: fixed;
  z-index: 1;
  left: 0;
  top: 80px;
  visibility: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F3F3F3;
  -webkit-animation: eTXFzq 0.15s linear;
  animation: eTXFzq 0.15s linear;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  border: 2px solid #EC7E00;
  border-top: 2px solid #D8D8D8;
  border-radius: 50%;
  width: 30px;
  height: 30px;
  -webkit-animation: fEWCgj 0.8s linear infinite;
  animation: fEWCgj 0.8s linear infinite;
}

@media only screen and (min-width:992px) {
  .c0 {
    min-height: 200px;
  }
}

<div
  class="c0"
  type="circle"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      mode="dark"
    />
  </div>
</div>
`;

exports[`Component: Loader should render as circle type and be showed 1`] = `
.c0 {
  width: 100%;
  height: calc(100% - 80px);
  position: fixed;
  z-index: 1;
  left: 0;
  top: 80px;
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F3F3F3;
  -webkit-animation: eTXFzq 0.15s linear;
  animation: eTXFzq 0.15s linear;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  border: 2px solid #EC7E00;
  border-top: 2px solid #D8D8D8;
  border-radius: 50%;
  width: 30px;
  height: 30px;
  -webkit-animation: fEWCgj 0.8s linear infinite;
  animation: fEWCgj 0.8s linear infinite;
}

@media only screen and (min-width:992px) {
  .c0 {
    min-height: 200px;
  }
}

<div
  class="c0"
  type="circle"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      mode="dark"
    />
  </div>
</div>
`;

exports[`Component: Loader should render as line type and be hidden 1`] = `
.c0 {
  width: 100%;
  height: calc(100% - 80px);
  position: fixed;
  z-index: 1;
  left: 0;
  top: 80px;
  visibility: hidden;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  border: 2px solid #EC7E00;
  border-top: 2px solid #D8D8D8;
  border-radius: 50%;
  width: 30px;
  height: 30px;
  -webkit-animation: fEWCgj 0.8s linear infinite;
  animation: fEWCgj 0.8s linear infinite;
}

<div
  class="c0"
  type="line"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      mode="dark"
    />
  </div>
</div>
`;

exports[`Component: Loader should render as line type and be showed 1`] = `
.c0 {
  width: 100%;
  height: calc(100% - 80px);
  position: fixed;
  z-index: 1;
  left: 0;
  top: 80px;
  visibility: visible;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  border: 2px solid #EC7E00;
  border-top: 2px solid #D8D8D8;
  border-radius: 50%;
  width: 30px;
  height: 30px;
  -webkit-animation: fEWCgj 0.8s linear infinite;
  animation: fEWCgj 0.8s linear infinite;
}

<div
  class="c0"
  type="line"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      mode="dark"
    />
  </div>
</div>
`;
